### PR TITLE
[DEMO] db support ph.I

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
* Added new scheme to db - `save_demo`
* Added `mysql` support for `save-demo`
* Added `git_repo` table to store git data
* Added `snapshot` table to version and executable name
* Added `tool` table to link `snapshot` and `git_repo`
* Added `dependency` table to store links between core `tool` and its dependencies

### CODE REVIEW

File: save-demo/db/tables/dependency.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        </createTable>
+
+        <addForeignKeyConstraint baseColumnNames="master_id"
+                                 baseTableName="dependency"
+                                 constraintName="fk_dependency_master"
+                                 referencedColumnNames="id"
+                                 referencedTableName="tool"
+                                 onDelete="CASCADE"/>
+
+        <addForeignKeyConstraint baseColumnNames="slave_id"
```
Comment (nulls) [inline save-demo/db/tables/dependency.xml:28]: can we change column name to `worker_id`, just for case.

File: save-demo/build.gradle.kts [hunk 2]
```diff
@@ -5,8 +5,16 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
...
     id("com.saveourtool.save.buildutils.spring-boot-app-configuration")
+    id("com.saveourtool.save.buildutils.spring-data-configuration")
     alias(libs.plugins.kotlin.plugin.serialization)
     kotlin("plugin.allopen")
+    alias(libs.plugins.kotlin.plugin.jpa)
+}
+
+kotlin {
+    allOpen {
+        annotation("javax.persistence.Entity")
```
Comment (nulls) [inline save-demo/build.gradle.kts:16]: I don't see entities classes

**Review by nulls — CHANGES_REQUESTED** [review]

File: save-demo/build.gradle.kts (hunk 2 above)
Comment (sanyavertolet) [reply save-demo/build.gradle.kts:16]: I have them in the `draft` branch, will be added to the next PR (trying to split it)

File: save-demo/db/tables/dependency.xml (hunk 1 above)
Comment (sanyavertolet) [reply save-demo/db/tables/dependency.xml]: Done

**Review by nulls — APPROVED** [review]

**Pull request merged**